### PR TITLE
Sol2

### PR DESCRIPTION
**Pull request by IPconfig**
### Solutions for Chapter 2

### CODE REVIEW

**Review by vrom911 — APPROVED** [review]
Amazing work! 👏🏼 
And the advanced task is cracked again 💯 

I wrote some suggestions, and what to do to make the CI pass for doctests... Sorry for the inconvenience with that!

I also tried to explain in steps the type tasks with the eta-reduction. Let me know if it make sense and if you would like to have one more task on that to challenge yourself on eta-reduction one more time 🙂

File: src/Chapter2.hs [hunk 1]
```diff
@@ -771,12 +781,14 @@ Let's now try to eta-reduce some of the functions and ensure that we
 mastered the skill of eta-reducing.
 -}
 divideTenBy :: Int -> Int
-divideTenBy x = div 10 x
+divideTenBy = div 10
 
 -- TODO: type ;)
-listElementsLessThan x l = filter (< x) l
+-- listElementsLessThan :: [[a]] -> [a]
+listElementsLessThan x y = filter (< x) y
```
[the remark this answers is not in the document]
Comment (IPconfig) [reply src/Chapter2.hs:788]: Thank you for the explanation! It is clear

File: src/Chapter2.hs [hunk 2]
```diff
@@ -136,34 +136,34 @@ functions in GHCi and insert the corresponding resulting output below:
 
 List of booleans:
 >>> :t [True, False]
-
+[True, False] :: [Bool]
```
[the remark this answers is not in the document]
Comment (vrom911) [reply src/Chapter2.hs:139]: @IPconfig , we fixed this in master.
If you want, you can rebase on master to get this fixed 🙂

**Review by vrom911 — APPROVED** [review]
Perfect! Congrats again! 🍾 

I am going to close this PR, so you can work further on other chapters 🙂  (Don't worry, this contribution won't disappear, and will be counted 👍🏼 )

**Pull request closed without merging**